### PR TITLE
Woo Email Confirmation: Adding design for the email verification step

### LINKED ISSUE #64162: Woo Email Confirmation - Implement design for interstitial step
Following on from https://github.com/Automattic/wp-calypso/issues/64159, implement the design for the new screen, as defined in DekZ88lDqG8Je04SyG1ZBK-fi-0%3A1.

image

### PR DESCRIPTION
**Pull request by valterlorran**
#### Proposed Changes

* Adds the design for the celebration step, which will ask for the user to verify their email.

Screen Shot 2022-06-08 at 16 53 29

#### Testing Instructions

* Go to `/setup/wooVerifyEmail?siteSlug=<YOUR-SITE>`

Closes #64162

### CODE REVIEW

**Review by sixhours — COMMENTED** [review]
Looking good so far!

This screen will also need a Back button (probably need to specify where `goBack` should go for the `emailVerifyCelebration` step in the setup flow file) and a "Confirm my email later" link in the upper right to skip the verification:

Screen Shot 2022-06-07 at 3 36 30 PM

"Confirm my email later" should go to `/home/[siteSlug]`.

Happy to leave those for a separate PR if you'd prefer to keep this one simpler!

We should also double-check the copy with design folks; "Please continue your journey from the following the link sent." is a confusing sentence...

**Comment by valterlorran** [comment]
Do you think that the text `Please continue your journey from the link sent.` would be more appropriate?

File: client/landing/stepper/declarative-flow/internals/steps-repository/woo-verify-email/index.tsx [hunk 1]
```diff
@@ -1,30 +1,77 @@
...
+import { useSelector } from 'react-redux';
 import FormattedHeader from 'calypso/components/formatted-header';
+import { useSiteSlugParam } from 'calypso/landing/stepper/hooks/use-site-slug-param';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { UserData } from 'calypso/lib/user/user';
+import { getCurrentUser } from 'calypso/state/current-user/selectors';
 import type { Step } from '../../types';
+import './style.scss';
+
+function redirect( to: string ) {
```
Comment (sixhours) [inline client/landing/stepper/declarative-flow/internals/steps-repository/woo-verify-email/index.tsx:15]: I know we've used this method elsewhere in the Stepper framework; should we make it a hook for easier use across steps/DRYer code?

**Review by sixhours — APPROVED** [review]
Looks great!

**Comment by sixhours** [comment]
> Please continue your journey from the link sent.

Sounds good to me!

File: client/landing/stepper/declarative-flow/internals/steps-repository/woo-verify-email/index.tsx (hunk 1 above)
Comment (valterlorran) [reply client/landing/stepper/declarative-flow/internals/steps-repository/woo-verify-email/index.tsx:15]: I will create an issue for that, ok?

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7366845

Thank you @valterlorran for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7366845).

**Pull request merged**